[yang-model] support reserving mem in high region for crashkernel

ysmanman opened the pull request:
<!--
     Please make sure you've read and understood our contributing guidelines:
     https://github.com/Azure/SONiC/blob/gh-pages/CONTRIBUTING.md

     ** Make sure all your commits include a signature generated with `git commit -s` **

     If this is a bug fix, make sure your description includes "fixes #xxxx", or
     "closes #xxxx" or "resolves #xxxx"

     Please provide the following information:
-->

#### Why I did it

By default, crashekernel is reserved in low mem region used by DMA. Support reserving high mem region for crashkernel to avoid occupying DMA mem.

##### Work item tracking
- Microsoft ADO **(number only)**:

#### How I did it

#### How to verify it

<!--
If PR needs to be backported, then the PR must be tested against the base branch and the earliest backport release branch and provide tested image version on these two branches. For example, if the PR is requested for master, 202211 and 202012, then the requester needs to provide test results on master and 202012.
-->

#### Which release branch to backport (provide reason below if selected)

<!--
- Note we only backport fixes to a release branch, *not* features!
- Please also provide a reason for the backporting below.
- e.g.
- [x] 202006
-->

- [ ] 202205
- [ ] 202211
- [ ] 202305
- [ ] 202311
- [ ] 202405
- [ ] 202411
- [ ] 202505

#### Tested branch (Please provide the tested image version)

<!--
- Please provide tested image version
- e.g.
- [x] 20201231.100
-->

- [ ] <!-- image version 1 -->
- [ ] <!-- image version 2 -->

#### Description for the changelog
<!--
Write a short (one line) summary that describes the changes in this
pull request for inclusion in the changelog:
-->

<!--
 Ensure to add label/tag for the feature raised. example - PR#2174 under sonic-utilities repo. where, Generic Config and Update feature has been labelled as GCU.
-->

#### Link to config_db schema for YANG module changes
<!--
Provide a link to config_db schema for the table for which YANG model
is defined
Link should point to correct section on https://github.com/Azure/sonic-buildimage/blob/master/src/sonic-yang-models/doc/Configuration.md
-->

#### A picture of a cute animal (not mandatory but encouraged)

ysmanman committed:
Use fixed crashkernel offset on OtterLake.

mssonicbld commented:
/azp run Azure.sonic-buildimage

ysmanman commented:
> @ysmanman can you update the testings done with this change.

Sure @deepak-singhal0408  Will do.

ysmanman committed:
Update yang model for configuring mem offset of crashkernel

ysmanman committed:
Upate yang model to allow setting offset for any range.

ysmanman commented on src/sonic-yang-models/yang-models/sonic-kdump.yang:
```diff
@@ -33,7 +33,7 @@ module sonic-kdump {
                 leaf memory {
                     type string {
-                        pattern "(((([0-9]+[MG]?)?(-([0-9]+[MG])?):)?[0-9]+[MG],?)+)";
+                        pattern "(((([0-9]+[MG]?)?(-([0-9]+[MG])?):)?[0-9]+[MG],?)+(@[0-9]+[MG])?)";
```
Good suggestion. Updated the change.

deepak-singhal0408 reviewed (dismissed):
LGTM. thanks Song.

ysmanman commented:
> @ysmanman please look at the PR checker failure.

Sure @deepak-singhal0408 . This is failure. Doesn't look like caused by my change. I will kick off build again.
 
```
2025-11-07T03:29:33.3468943Z Error type: RUN_TEST_CASE_FAILED
2025-11-07T03:29:33.3469268Z Error message: route/test_duplicate_route.py|||2 failed for RUN_TEST_CASE_FAILED, stop whole test plan
2025-11-07T03:29:33.3469821Z Operation failed with exception: Exception('Test plan id: 690d4ca98ad8ec69c27e4924, status: FAILED, result: FAILED, Elapsed 5438 seconds. Check ```

ysmanman commented:
/AzurePipelines run Azure.sonic-buildimage

ysmanman commented:
/azpw run Azure.sonic-buildimage

ysmanman committed:
Support reserving mem in high region for crashkernel

judyjoseph commented on files/Aboot/boot0.j2:
```diff
@@ -488,7 +488,12 @@ get_eeprom_value() {
-        Wolverine*|Clearwater2*|OtterLake*|QuartzDd*|Redstart8Mk2Quartz4*|CitrineDd*)
+        OtterLake*)
+            if ! cmdline_has crashkernel; then
+                cmdline_add crashkernel=1G,high
```
general question, is this 1G high applicable only for this platform. Should me make crashkernel allocation into high memory default ?

Also I see another PR from Arista : https://github.com/sonic-net/sonic-buildimage/pull/24382. May be we should close one of them ?

ysmanman committed:
The committed change includes both platform-specific and platform-independent changes.
The plan is to split the change into two separate PRs. Revert platform change and will
create a new PR for platform change.

deepak-singhal0408 commented on src/sonic-yang-models/tests/yang_model_tests/tests_config/kdump.json:
```diff
@@ -55,27 +55,27 @@
         }
     },
-    "KDUMP_WITH_VALID_MEM_OFFSET": {
+    "KDUMP_WITH_VALID_MEM_HIGH": {
```
got it. thanks

mssonicbld commented:
/AzurePipelines run Azure.sonic-buildimage

mssonicbld commented:
/AzurePipelines run Azure.sonic-buildimage

mssonicbld commented:
/AzurePipelines run Azure.sonic-buildimage

ysmanman committed:
Merge branch 'sonic-net:master' into crashkernel-offset

ysmanman committed:
Merge branch 'sonic-net:master' into crashkernel-offset

ysmanman committed:
Merge branch 'sonic-net:master' into crashkernel-offset

mssonicbld commented:
/azp run Azure.sonic-buildimage

ysmanman committed:
Merge branch 'sonic-net:master' into crashkernel-offset

ysmanman committed:
Remove unused test case

mssonicbld commented:
/azp run Azure.sonic-buildimage

rlhui commented:
@ysmanman  please rebase? thanks.

ysmanman committed:
Merge branch 'master' into crashkernel-offset

mssonicbld commented:
/AzurePipelines run Azure.sonic-buildimage

ysmanman commented:
/azpw run Azure.sonic-buildimage

mssonicbld commented:
⚠️ **Notice**: `/azpw run` only runs failed jobs now. If you want to trigger a whole pipline run, please rebase your branch or close and reopen the PR.
💡 **Tip**: You can also use `/azpw retry` to retry failed jobs directly.

Retrying failed(or canceled) jobs...

mssonicbld commented:
Build not found. Please close and reopen the PR or rebase your branch to trigger a new build.

arlakshm commented:
/Azp run

arlakshm commented:
@ysmanman, can you please sign the commits for the DCO checker to pass
